### PR TITLE
Fix find_project_root with relative paths

### PR DESCRIPTION
**Pull request by andersk**
The `common_path` crate doesn’t work correctly with relative paths: `common_path("a/b", "c/d") = None`. (I would have reported this upstream to `common_path`, but its [repository](https://gitlab.com/pwoolcoc/common-path) doesn’t exist anymore.)

Additionally, we want to be able to find a parent of the current directory, but `.ancestors()` will not ascend past `"."`.

Work around both problems by converting paths to absolute.

### CODE REVIEW

File: src/pyproject.rs [hunk 1]
```diff
@@ -69,7 +70,10 @@ fn find_user_pyproject_toml() -> Option<PathBuf> {
 }
 
 fn find_project_root(sources: &[PathBuf]) -> Option<PathBuf> {
-    if let Some(prefix) = common_path_all(sources.iter().map(PathBuf::as_path)) {
+    let cwd = current_dir().unwrap_or_else(|_| ".".into());
+    // common_path doesn't work correctly with relative paths
+    let absolute_sources: Vec<PathBuf> = sources.iter().map(|source| cwd.join(source)).collect();
```
Comment (Stranger6667) [inline src/pyproject.rs:75]: Nit: `Vec` allocation is not necessary, as `common_path_all` accepts an iterator. I suggest merging `map` calls:

```rust
sources.iter().map(|source| cwd.join(source).as_path())
```

Comment (andersk) [reply src/pyproject.rs:75]: `common_path_all` accepts an iterator of `Path` references. Your suggestion doesn’t compile as the temporary `PathBuf` isn’t in scope long enough.

Comment (Stranger6667) [reply src/pyproject.rs]: Oh indeed, I missed that. Sorry!

**Comment by charliermarsh** [comment]
Nice, this looks reasonable. Do you know what problems this was causing in practice?

(Also, in the given example of `common_path("a/b", "c/d") = None` -- what's the "correct" return value there? "."?)

**Comment by andersk** [comment]
If you have a project with `pyproject.toml`, `a/b.py`, `c/d.py`, then previously

* `ruff a/b.py c/d.py` failed to respect any of the settings in `pyproject.toml`;
* `cd a; ruff b.py` failed to respect any of the settings in `pyproject.toml`.

`common_path("a/b", "c/d") = Some(".")` seems like it’d be more correct, and would have fixed the first of these points but not the second.

**Comment by charliermarsh** [comment]
This makes sense, thank you!

**Pull request merged**